dump/plugins/portchannel.py

```diff
@@ -0,0 +1,130 @@
+    
+    def init_lag_member_appl_info(self):
+        req = MatchRequest(db="APPL_DB", table="LAG_MEMBER_TABLE", key_pattern=self.lag_name+":*", ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        for key in ret["keys"]:
+            self.lag_members.add(key.split(":")[-1])
+        self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"])
+        
+    def init_lag_state_info(self):
+        req = MatchRequest(db="STATE_DB", table="LAG_TABLE", key_pattern=self.lag_name, ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        return self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"])
+    
+    def init_lag_member_state_info(self):
+        req = MatchRequest(db="STATE_DB", table="LAG_MEMBER_TABLE", key_pattern=self.lag_name+"|*", ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        for key in ret["keys"]:
+            self.lag_members.add(key.split("|")[-1])
+        self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"])
+    
+    def init_lag_asic_info(self, lag_type_objs_asic):
+        if len(lag_type_objs_asic) == 0:
+            self.ret_temp["ASIC_DB"]["tables_not_found"].extend(["ASIC_STATE:SAI_OBJECT_TYPE_LAG"])
+            return 
+        for lag_asic_obj in lag_type_objs_asic:
+            req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_LAG", key_pattern=lag_asic_obj, ns=self.ns)
+            ret = self.match_engine.fetch(req)
+            self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"])
+    
+    def init_lag_member_type_obj_asic_info(self):
+        lag_type_objs_asic = set()
+        for port_name in self.lag_members:
+            port_asic_obj = self.get_port_asic_obj(port_name)
+            if port_asic_obj:
+                lag_member_key, lag_oid = self.get_lag_member(port_asic_obj)
+                self.add_to_ret_template("ASIC_STATE:SAI_OBJECT_TYPE_LAG_MEMBER" , "ASIC_DB", [lag_member_key], "")
+                lag_type_objs_asic.add(lag_oid)
+        if not lag_type_objs_asic:
+            self.ret_temp["ASIC_DB"]["tables_not_found"].extend(["ASIC_STATE:SAI_OBJECT_TYPE_LAG_MEMBER"])
+        return lag_type_objs_asic
+             
+    def get_port_asic_obj(self, port_name):
+        req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_HOSTIF", key_pattern="*", field="SAI_HOSTIF_ATTR_NAME", 
+                           value=port_name, return_fields=["SAI_HOSTIF_ATTR_OBJ_ID"], ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        asic_port_obj_id = ""
+        if not ret["error"] and ret["keys"]:
+            sai_hostif_obj_key = ret["keys"][-1]
+            if sai_hostif_obj_key in ret["return_values"] and "SAI_HOSTIF_ATTR_OBJ_ID" in ret["return_values"][sai_hostif_obj_key]:
+                asic_port_obj_id = ret["return_values"][sai_hostif_obj_key]["SAI_HOSTIF_ATTR_OBJ_ID"]
+        return asic_port_obj_id
+    
+    def get_lag_member(self, port_asic_obj):
```

Comment:
Since it returns both lag and member oid can we rename the function to get_lag_and_member_obj?

Reply:
Done!